### PR TITLE
Strict parsing of configuration files

### PR DESCRIPTION
**Pull request by divergentdave**
This PR implements one half of #2195. I added `#[serde(deny_unknown_fields)]` to all configuration types, which will cause an error if there are any unexpected keys in a table. This turned up one issue in the example forwarder config file.

I also added a test that starts with all config files in `tests/test-data/test_configs`, mutates them by adding an extra key to each table in turn, and confirms that it gets an "unknown field" error when trying to parse each mutated input. Since `#[serde(deny_unknown_fields)]` is configured on a type-by-type basis, it can be easy to forget, so this test is intended to help maintain this behavior going forward. It can't discover new tables on its own, but so long as the example/test configuration files have good coverage, it should help.

### CODE REVIEW

File: crates/server/tests/config_tests.rs [hunk 1]
```diff
@@ -273,3 +279,225 @@ define_test_config!(openssl_dnssec);
...
+        let mut skip = false;
+        #[cfg(not(feature = "dnssec"))]
+        if config_table.contains_key("tls_cert") {
+            println!("skipping due to tls_cert setting");
+            skip = true;
+        }
+        if let Some(zones) = config_table.get("zones") {
+            let zones = zones.as_array().unwrap();
+            for zone in zones {
+                if let Some(stores) = zone.get("stores") {
```
Comment (djc) [inline crates/server/tests/config_tests.rs:453]: Nit: suggest doing an `let Some(stores) = zones.get("stores") else { continue }` here to reduce rightward drift.

**Review by djc — APPROVED** [review]
LGTM, thanks!

File: crates/server/tests/config_tests.rs [hunk 2]
```diff
@@ -447,36 +447,35 @@ fn test_reject_unknown_fields() {
...
-                    let stores = stores.as_table().unwrap();
-                    let _store_type = stores.get("type").unwrap().as_str().unwrap();
-
-                    #[cfg(not(feature = "sqlite"))]
-                    if _store_type == "sqlite" {
-                        println!("skipping due to sqlite store");
-                        skip = true;
-                        break;
-                    }
+        let zones = config_table.get("zones").unwrap().as_array().unwrap();
```
Comment (djc) [inline crates/server/tests/config_tests.rs:450]: Please squash this into the originating commit.

**Review by djc — APPROVED** [review]

**Review by bluejekyll — APPROVED** [review]
This looks like a great change, thanks for adding this.

**Pull request merged**